File: LeetCode/C++/1000-1099/1047.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        stack<char> st;
        for(int i = 0 ; i < S.length();++i){
            if(st.empty() || st.top() != S[i]){
                st.push(S[i]);
            }
            else{
                st.pop();
            }
        }
        
        string ans = "";
        while(!st.empty()){
            ans += st.top();
            st.pop();
        }
        
        reverse(ans.begin(), ans.end());
        
        return ans;
    }
};

File: LeetCode/C#/1000-1099/1047.cs
public class Solution {
    private static string ReverseString(string s)
    {
        char[] array = s.ToCharArray();
        Array.Reverse(array);
        return new string(array);
    }
    
    public string RemoveDuplicates(string S) {
        Stack<char> st = new Stack<char>();
        for(int i = 0 ; i < S.Length;++i){
            if(st.Count == 0 || st.Peek() != S[i]){
                st.Push(S[i]);
            }
            else{
                st.Pop();
            }
        }
        
        StringBuilder ans = new StringBuilder();
        while(st.Count > 0){
            ans.Append(st.Peek());
            st.Pop();
        }
        
        return ReverseString(ans.ToString());
    }
}